web/static/js/components/group_label_input.jsx

```diff
@@ -7,23 +7,63 @@ import styles from "./css_modules/group_label_input.css"
 // exceed the 1/4th column width on desktop, even with the widest possible characters
 const MAX_LENGTH_OF_GROUP_NAME = 20
 
-const GroupLabelInput = ({ groupWithAssociatedIdeasAndVotes, actions }) => (
-  <div className="ui fluid input">
-    <input
-      type="text"
-      className={styles.textInput}
-      placeholder="Add a group label"
-      defaultValue={groupWithAssociatedIdeasAndVotes.label}
-      maxLength={MAX_LENGTH_OF_GROUP_NAME}
-      onBlur={e => {
-        actions.submitGroupLabelChanges(groupWithAssociatedIdeasAndVotes, e.target.value)
-      }}
-    />
-    <div className="instruction">
-      <span className="keyboard-key">tab</span> to submit
-    </div>
-  </div>
-)
+class GroupLabelInput extends Component {
+  state = (() => {
+    const { groupWithAssociatedIdeasAndVotes } = this.props
+    const { label } = groupWithAssociatedIdeasAndVotes
+    return {
+      oldLabel: label,
+      showCheckmarkForLabelId: null,
+    }
+  })();
+
+  static getDerivedStateFromProps(props, state) {
+    const { oldLabel } = state
+    const { groupWithAssociatedIdeasAndVotes } = props
+    const newLabel = groupWithAssociatedIdeasAndVotes.label
+    if (oldLabel !== newLabel) {
+      return ({
+        oldLabel: newLabel,
+        showCheckmarkForLabelId: groupWithAssociatedIdeasAndVotes.id })
+    }
+    return null
+  }
+
+  labelUpdatedCheckmark = () => {
+    const classes = ["check icon"]
+    classes.push(styles.updateSucceededCheckmark)
```

Comment:
given that `styles.updateSucceededCheckmark` maps to a string, we can remove the `Array.prototype.push` call in favor of an array literal like 

```es6
const classes = ["check icon", styles.updateSucceededCheckmark]
```

Reply:
Cool, didn't know that.  Modifying as per suggestion.